ui/components/app/name/name-details/name-details.tsx

```diff
@@ -96,7 +96,9 @@ function getProviderLabel(
   if (LOCALIZED_PROVIDERS.includes(sourceId)) {
     // Use intermediate variable to avoid "Forbidden use of template strings
     // in 't' function" error.
-    const messageKey = `nameProvider_${sourceId}`;
+    const messageKey = `nameProvider_${
+      sourceId as 'ens' | 'etherscan' | 'lens' | 'token'
+    }` as const;
     return t(messageKey);
```

Comment:
Just a quirk due to the translations being a bit more strict - it doesn't accept any arbitrary string.
Meh we can loosen it or have a separate type definition.

In fact could slowly migrate from a loose function definition to a strict function definition if we wanted to.